File: 卡码/代码随想录/1/08-贪心算法/Go/01-简单/LeetCode_1005.go
// 1005. K 次取反后最大化的数组和
package main

import (
	"math"
	"sort"
)

func largestSumAfterKNegations(nums []int, k int) int {
	sort.Slice(nums,func(i,j int) bool{
		return math.Abs(float64(nums[i]))>math.Abs(float64(nums[j]))
	})
	res:=0
	for i:=0;i<len(nums);i++ {
		if k>0&&nums[i]<0 {
			nums[i]*=-1
			k--
		}
		res+=nums[i]
	}
	if k%2==1 {
		res-=2*nums[len(nums)-1]
	}
	return res
}

File: 卡码/代码随想录/1/08-贪心算法/Js/01-简单/LeetCode_1005.js
//1005. K 次取反后最大化的数组和
const largestSumAfterKNegations = function(nums, k) {
  nums.sort((a,b)=>Math.abs(b)-Math.abs(a));
  let res=0;
  for(let i=0;i<nums.length;i++){
    if(nums[i]<0&&k>0){
      nums[i]=-nums[i];
      k--;
    }
    res+=nums[i];
  }
  if(k%2==1) res-=2*nums[nums.length-1];
  return res;
};